support-lib/objc/DJIMarshal+Private.h

```diff
@@ -118,6 +118,27 @@ struct String {
     }
 };
 
+struct WString {
+    using CppType = std::wstring;
+    using ObjcType = NSString*;
+
+    using Boxed = WString;
+
+    static CppType toCpp(ObjcType string) {
+        assert(string);
+        NSStringEncoding encoding = CFStringConvertEncodingToNSStringEncoding(kCFStringEncodingUTF32LE);
+        NSData* data = [string dataUsingEncoding:encoding];
+        return std::wstring((wchar_t*)[data bytes], [data length] / sizeof (wchar_t));
+    }
+
+    static ObjcType fromCpp(const CppType& string) {
+        assert(string.size() <= std::numeric_limits<NSUInteger>::max());
+        return [[NSString alloc] initWithBytes:string.data()
+                                        length:string.size() * sizeof(wchar_t)
+                                      encoding:NSUTF32LittleEndianStringEncoding];
```

Comment:
I don't understand why the encoding constants here are fixed to UTF32.  Shouldn't there need to be some variability between UTF16 and UTF32 as there is in Java?  Are you assuming that whcar_t is always 32-bits in Objective-C?  That assumption seems to be sound for iOS and OSX, though I'm concerned it might not be truly universal, or unchangeable.  I think the code here should at least have a static_assert to check that assumption, if you're not going to go so far as to implement the 16-bit option.